tests/testutil/service.go

```diff
@@ -202,6 +209,88 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+// GetEndpointUrl returns the URL for this service's app if the App object is versioned it returns
+// the versioned url.
+func (ts *TestService) GetEndpointUrl() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+// GetKubeEndpointUrlByRevision returns the versioned endpoint URL by using the revision provided
+// and returns it as string
+func (ts *TestService) GetKubeEndpointUrlByRevision(revision string) string {
```

Comment:
Don't need revision here, that lives on ts.Name. See above comment as this method will need to change anyway.